dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/business/FileAssetFactoryImpl.java

```diff
@@ -35,45 +38,59 @@ public FileAssetFactoryImpl() {
         permissionAPI = APILocator.getPermissionAPI();
     }
 
-    @Override
-    public List<Contentlet> findFileAssetsByFolderInDB(
-            final Folder parentFolder,
-            final  User user,
-            final boolean live) throws DotDataException, DotSecurityException {
-
-        final boolean respectFrontendRoles = live;
+    
+    /**
+     * takes a list of inodes and returns the list of contentlets - includes permissions check
+     * 
+     * @param identifiers
+     * @param user
+     * @param respectFrontendRoles
+     * @return
+     */
 
-        if(!permissionAPI.doesUserHavePermission(parentFolder, PermissionAPI.PERMISSION_READ, user, respectFrontendRoles)){
-            throw new DotSecurityException("User:" + user.getUserId() + " does not have permissions on Folder " + parentFolder);
-        }
+    @NotNull
+    List<Contentlet> fromIdentifiers(
+            final List<String> identifiers,
+            final User user,
+            final boolean respectFrontendRoles
+            )  {
 
-        final DotConnect dotConnect = new DotConnect();
-        dotConnect.setSQL(String.format(FIND_FILE_ASSETS_BY_FOLDER_QUERY, live ? "live" : "working"));
-        dotConnect.addParam(parentFolder.getPath());
-        dotConnect.addParam(parentFolder.getHostId());
+        final ImmutableList.Builder<Contentlet> builder = new ImmutableList.Builder<>();
 
-        final List<Map<String, Object>> queryResults = dotConnect.loadObjectResults();
-        return convertToFileAssets(user, respectFrontendRoles, queryResults);
+        for (final String  fileInfo : identifiers) {
+            final Contentlet file = Try.of(()-> APILocator.getContentletAPI().find(fileInfo, user, respectFrontendRoles)).getOrNull();
+            if(file==null)continue;
```

Comment:
Codacy found an issue: [This statement should have braces](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6325390)